src/gluonts/shell/serve.py

```diff
@@ -207,4 +207,73 @@ def predictor_factory(request) -> Predictor:
         },
     )
 
-    gunicorn_app.run()
+    return gunicorn_app
+
+
+class ServerFacade:
+    """
+    A convenience wrapper for sending requests and handling responses to
+    an inference server located at the given address.
+    """
+
+    @validated()
+    def __init__(self, base_address: str) -> None:
+        self.base_address = base_address
+
+    def url(self, *parts) -> str:
+        return "/".join((self.base_address,) + parts)
+
+    def ping(self) -> bool:
+        try:
+            response = requests.get(
+                url=self.url("ping"),
+                headers={
+                    "Content-Type": "application/json",
+                    "Accept": "application/json",
+                },
+            )
+            return response.status_code == 200
+        except requests.exceptions.ConnectionError:
+            return False
+
+    def execution_parameters(self) -> dict:
+        response = requests.get(
+            url=self.url("execution-parameters"),
+            headers={
+                "Content-Type": "application/json",
+                "Accept": "application/json",
+            },
+        )
+
+        response_code = response.status_code
+
+        if response_code == 200:
+            return response.json()
+        elif response_code >= 400:
+            raise RuntimeError(response.content.decode("utf-8"))
+        else:
+            raise RuntimeError(f"Unexpected response code {response_code}")
+
+    def invocations(
+        self, data_entries: Iterable[DataEntry], configuration: dict
+    ) -> List[dict]:
+        instances = list(map(serialize_data_entry, data_entries))
+        response = requests.post(
+            url=self.url("invocations"),
+            json={"instances": instances, "configuration": configuration},
+            headers={
+                "Content-Type": "application/json",
```

Comment:
Why do you set these explicitly? `"Content-Type": "application/json"` is set automatically, if you use the `json` arg.

Reply:
I prefer to be more explicit here for code readability, as the SageMaker contract expects the "Content-Type" and "Accept" headers to be present.